sdk/src/main/java/com/mapbox/maps/renderer/MapboxRenderThread.kt

```diff
@@ -272,30 +281,50 @@ internal class MapboxRenderThread : Choreographer.FrameCallback {
 
   @WorkerThread
   override fun doFrame(frameTimeNanos: Long) {
-    // if frame started being rendered time is less or equal to previous one
-    // it means we did not have time to render everything on previous frame
-    // we do no drawing in order not to overload render thread
-    if (frameTimeNanos <= previousVsyncFrameStartTimeNs) {
-      return
-    }
-    if (eglPrepared && !paused && !shouldExit) {
-      previousVsyncFrameStartTimeNs = frameTimeNanos
-      draw()
+    try {
+      awaitingNextVsync.set(false)
+      // if frame started being rendered time is less or equal to previous one
+      // it means we did not have time to render everything on previous frame
+      // we do no drawing in order not to overload render thread
+      if (eglPrepared && !paused && !shouldExit) {
+        draw()
+      }
+    } finally {
+      // regardless if we did drawing or not execute all non gl relative events
+      eventQueue.apply {
+        if (isNotEmpty()) {
+          forEach {
+            it.run()
+          }
+          clear()
+        }
+      }
```

Comment:
nit: duplicate code for iterating a runnable queue and running it, could be refactored into 1 function that takes either the `eventQueu` or `renderEventQueue`